src/shims/unix/android/foreign_items.rs

```diff
@@ -25,8 +26,65 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(errno_place.to_ref(this).to_scalar(), dest)?;
             }
 
+            // Threading
+            "prctl" => {
+                // We do not use `check_shim` here because `prctl` is variadic. The argument
+                // count is checked bellow.
+                this.check_abi_and_shim_symbol_clash(abi, Abi::C { unwind: false }, link_name)?;
+
+                check_args_len("prctl", args, 1)?;
+
+                let id = this.read_scalar(&args[0])?.to_i32()?;
+                // FIXME: Use PR_SET_NAME and PR_GET_NAME constants when
+                // https://github.com/rust-lang/libc/pull/3941 lands.
+                let pr_set_name = 15_i32;
+                let pr_get_name = 16_i32;
+
+                let res = match id {
+                    id if id == pr_set_name => {
+                        check_args_len("'PR_SET_NAME' prctl", args, 2)?;
+
+                        let tid = this.linux_gettid()?;
```

Comment:
The problem seems to be in this line, it works with ``let tid = this.pthread_self()?;``. 

The test report UB because of the size mismatch in this line:
https://github.com/rust-lang/miri/blob/bf0801d3d90e3077e2fd7bf4eaf913a4a499b5dc/src/shims/unix/thread.rs#L75